### PR TITLE
HOCS-1076-Search for a compensation case

### PR DESCRIPTION
**Pull request by hubert-raich**

### CODE REVIEW

File: server/routes/search.js [hunk 1]
```diff
@@ -14,7 +14,10 @@ router.all(['/search', '/api/search', '/api/form/search'], getForm(form, { submi
 router.get('/api/form/search', (req, res) => res.json(req.form));
 
 router.post(['/search/results', '/api/search/results'],
-    processRequestBody(form().getFields()),
+    async (req, res, next) => {
```
Comment (lewismuirhead) [inline server/routes/search.js]: this could be abstracted into a middleware component

File: server/routes/search.js [hunk 2]
```diff
@@ -33,7 +36,10 @@ router.post(['/search/results', '/api/search/results'],
                 correspondentName: formData['correspondent'],
                 topic: formData['topic'],
                 data: {
-                    POTeamUUID: formData['signOffMinister']
+                    POTeamUUID: formData['signOffMinister'],
```
Comment (lewismuirhead) [inline server/routes/search.js]: mainly existing code, but I think this handler should definitely be in a middleware with some tests

File: server/services/action.js [hunk 3]
```diff
@@ -86,15 +86,16 @@ const actions = {
...
+                        const documentTags = (await listServiceInstance.fetch('S_SYSTEM_CONFIGURATION')).documentLabels;
+                        response = await createCase('/case', { caseType: context, form }, documentTags[0], headers);
                         clientResponse = { summary: 'Created a new case ', link: `${response.data.reference}` };
                         return handleActionSuccess(clientResponse, workflow, form);
                     }
                     case actionTypes.BULK_CREATE_CASE: {
                         const listServiceInstance = listService.getInstance(uuid(), null);
-                        const documentTags = await listServiceInstance.fetch('S_DOCUMENT_TAGS');
-                        response = await createCase('/case/bulk', { caseType: context, form }, documentTags[0].value, headers);
+                        const documentTags = (await listServiceInstance.fetch('S_SYSTEM_CONFIGURATION')).documentLabels;
```
Comment (lewismuirhead) [inline server/services/action.js]: I think: const { documentLabels: documentTags } = await listServiceInstance....
would remove the need for the brackets

File: server/routes/search.js (hunk 2 above)
Comment (hubert-raich) [reply server/routes/search.js]: resolved with searchHandler component

File: server/routes/search.js (hunk 1 above)
Comment (hubert-raich) [reply server/routes/search.js]: resolved with searchHandler component

File: server/services/action.js (hunk 3 above)
Comment (hubert-raich) [reply server/services/action.js]: done

**Pull request merged**